### PR TITLE
Cleanup ReadTheDocs build and deploy master branch

### LINKED ISSUE #3205: Deploy master branch docs to RTD
I noticed the `geotrellis-slick` module is now included in `geotrellis-contrib` and the documentation does not reflect this https://geotrellis.readthedocs.io/en/latest/guide/module-hierarchy.html

I assume other docs are out of date as well and the new docs just need to be deployed

### PR DESCRIPTION
**Pull request by CloudNiner**
# Overview

It looks like the core issue in #3205 is addressed simply by pushing a button in ReadTheDocs (RTD). I declined to automate ReadTheDocs publishes from master since we plan to move away from RTD. We can push the button a few more times. But I did add a reminder to publish RTD in our github release issue template. 

The PR fixes up a few remaining warnings printed by `make html`  when building the docs with sphinx and ensures that some pages that lost their links have them back.

## Checklist

- [x] [./CHANGELOG.md](https://github.com/locationtech/geotrellis/blob/master/CHANGELOG.md) updated, if necessary. Link to the issue if closed, otherwise the PR.
- [x] [Module Hierarchy](https://github.com/locationtech/geotrellis/blob/master/docs/guide/module-hierarchy.rst) updated, if necessary
- [x] `docs` guides update, if necessary
- [x] New user API has useful Scaladoc strings
- [x] Unit tests added for bug-fix or new feature

## Demo

Screen Shot 2020-12-08 at 3 56 39 PM

Closes #3205

### CODE REVIEW

**Comment by CloudNiner** [comment]
@pomadchin the two remaining warnings are due to invalid SQL code [here](https://github.com/locationtech/geotrellis/blame/master/docs/cassandra/cassandra-optimization.rst#L45) and [here](https://github.com/locationtech/geotrellis/blame/master/docs/cassandra/cassandra-optimization.rst#L89). Any chance you know what those statements should actually be so that they're valid SQL?

**Review by jpolchlo — APPROVED** [review]
LGTM.

Death to ETL!!!

File: docs/architecture/004-etl-pipeline.rst [hunk 1]
```diff
@@ -1,412 +0,0 @@
-0004 - ETL Pipelne
```
Comment (pomadchin) [inline docs/architecture/004-etl-pipeline.rst:1]: Actually that is a decription of this module: https://github.com/locationtech/geotrellis/tree/master/spark-pipeline/src

Which inspired by https://pdal.io/quickstart.html

Which leads us closer to resolving https://github.com/locationtech/geotrellis/issues/1302 if we want it. 

Should not we remove module first than?

Comment (CloudNiner) [reply docs/architecture/004-etl-pipeline.rst:1]: This is a duplicate file of https://github.com/locationtech/geotrellis/blob/master/docs/architecture/005-etl-pipeline.rst

It remains in the docs: https://geotrellis.readthedocs.io/en/latest/architecture/005-etl-pipeline.html

Comment (pomadchin) [reply docs/architecture/004-etl-pipeline.rst:1]: Ouch haven't noticed that, thanks!

**Pull request merged**